plugins/pulp_docker/plugins/importers/upload.py

```diff
@@ -148,6 +170,46 @@ def get_models(self, metadata, mask_id=''):
         return images
 
 
+class ProcessManifest(PluginStep):
+    """
+    Retrieves image manifest from an uploaded tarball and pull out the
+    metadata for further processing
+    """
+
+    def process_main(self, item=None):
+        """
+        Pull the image manifest out of the tar file
+
+        :param item: Not used by this Step
+        :type item: None
+        """
+        image_manifest = json.dumps(tarutils.get_image_manifest(self.parent.file_path))
+        digest = models.UnitMixin.calculate_digest(image_manifest)
+        with open(os.path.join(self.get_working_dir(), digest), 'w') as manifest_file:
+            manifest_file.write(image_manifest)
+        manifest = models.Manifest.from_json(image_manifest, digest)
+        self.parent.available_units.append(manifest)
+        self.parent.available_units.extend(self.get_models(manifest))
+
+    def get_models(self, manifest):
+        """
+        Given an image manifest, returns model instances to represent each blob of the image defined
+        by the unit_key.
+
+        :param manifest: An initialized Manifest object
+        :type manifest:  pulp_docker.plugins.models.Manifest
+        :return:         list of models.Blob instances
+        :rtype:          list
+        """
+        available_blobs = set()
+        for layer in manifest.fs_layers:
+            available_blobs.add(layer.blob_sum)
```

Comment:
i found out that skopeo downloads foreign blobs, we do not want to save those into pulp.
we need to add here similar check as we have it in sync- i.e. check the type of the layer and in case it is foreign do not add it to the available blobs.